mcf-browser/src/back-end/parse-tmcf.ts

```diff
@@ -245,131 +198,116 @@ class ParseTmcf {
+ * @param {Array<CsvRow>} csvRows The json representation of the csv file.
+ *     Each CsvRow element of the array represents one row of the csv.
+ * @return {ParsedCsv} An object containing the parsed datapoints and mcf
+ * strings
+ */
+  parseCsvRows(template: string, csvRows: CsvRow[]) : ParsedCsv {
     this.csvIndex = 1;
 
+    // Split tmcf file into StatVarObs entities and non-StatVarObs entities
+    const {dataEntities, otherEntities} =
+        ParseTmcf.getEntityTemplates(template);
+
     const datapoints: TimeDataObject = {};
-    const entityTemplates = this.getEntityTemplates(template);
+    const otherMcfs = otherEntities;
+
     for (const row of csvRows) {
-      for (const entityTemplate of entityTemplates) {
+      for (const entityTemplate of dataEntities) {
         const parsedCsvRow = this.getFacetAndValueFromRow(
             entityTemplate,
             row,
         );
 
         if (parsedCsvRow) {
-          const {facet, date, value} = parsedCsvRow;
+          const {facet, date, value, mcf} = parsedCsvRow;
           const parsedValue = value !== '' ? parseFloat(value) : undefined;
-          datapoints[facet] = datapoints[facet] ?
-              {
-                ...datapoints[facet],
-                [date]: parsedValue,
-              } :
-              {[date]: parsedValue};
+          const datapoint = {
+            mcf,
+            value: parsedValue,
+          };
+
+          // Add
+          const seriesObject = datapoints[facet] ?
+              datapoints[facet] as SeriesObject : {};
+          seriesObject[date] = datapoint;
+          datapoints[facet] = seriesObject;
         }
       }
       this.csvIndex += 1;
     }
-    return datapoints;
+    return {datapoints, otherMcfs};
   }
 
   /**
-   * Converts CSV file to a JS object where the keys are facet and the
-   * values are the datapoints
-   * @param {string} template The string representation of a tmcf file.
-   * @param {FileObject} csvFile The csv file from html file-input element.
-   * @return {TimeDataObject} The json representation of the csv file.
-   */
-  async getDataPointsFromFile(
-      template: string, csvFile: Blob,
-  ) : Promise<TimeDataObject> {
-    const fileReader = new FileReader();
-    fileReader.readAsText(csvFile);
-    return new Promise((res, rej) => {
-      fileReader.addEventListener('loadend', () => {
-        const csv = require('csvtojson');
-        csv()
-            .fromString(fileReader.result)
-            .then((csvRows: CsvRow[]) => {
-              res(this.csvToDataPoint(template, csvRows));
-            });
-      });
-      fileReader.addEventListener('error', rej);
+  * Turns a CSV string array to a CsvRow object
+  * @param {string[]} lines an array of lines where each line is a row
+  * of the csv
+  * @return {CsvRow[]} an array of objects with keys as column names and
+  * values as the corresponding row value
+  */
+  static convertCsvToJson(lines: string[]) : CsvRow[] {
```

Comment:
have you tested this csv parsing function w/:
- cell values that have a comma e.g., if a cell has "abc, bcd"
- cell values that have double quotes